eventuate-tram-consumer-kafka/src/main/java/io/eventuate/tram/consumer/kafka/MessageConsumerKafkaImpl.java

```diff
@@ -34,32 +35,35 @@ public MessageConsumerKafkaImpl(String bootstrapServers) {
   @Autowired
   private DuplicateMessageDetector duplicateMessageDetector;
 
+
   @Override
   public void subscribe(String subscriberId, Set<String> channels, MessageHandler handler) {
-    BiConsumer<ConsumerRecord<String, String>, BiConsumer<Void, Throwable>> kcHandler = (record, callback) -> {
-      Message m = toMessage(record);
+    SwimlaneBasedDispatcher swimlaneBasedDispatcher = new SwimlaneBasedDispatcher(subscriberId, Executors.newCachedThreadPool());
 
-      // TODO If we do that here then remove TT from higher-levels
+    BiConsumer<ConsumerRecord<String, String>, BiConsumer<Void, Throwable>> kcHandler = (record, callback) -> {
+      swimlaneBasedDispatcher.dispatch(toMessage(record), record.key().hashCode() % 8, message ->
```

Comment:
The swimlane should be `ConsumerRecord.partition` - NOT `record.key().hashCode() % 8`

Reply:
ok